src/main/java/io/github/moulberry/notenoughupdates/miscgui/AccessoryBagOverlay.java

```diff
@@ -1059,4 +901,179 @@ public static int getRarity(ItemStack stack) {
+		if (info == null) return bin;
+		if (bin == -1) return info.craftCost;
+		return Math.min(info.craftCost, bin);
+	}
+
+	public static ScaledResolution getScaledResolution() {
+		return new ScaledResolution(Minecraft.getMinecraft());
+	}
+
+	public static int mouseX() {
+		return Mouse.getX() / getScaledResolution().getScaleFactor();
+	}
+
+	public static int mouseY() {
+		return getScaledResolution().getScaledHeight() - Mouse.getY() / getScaledResolution().getScaleFactor();
+	}
+
+	public static int getMagicalPowerForItem(String internal) {
+		int abi = 0;
+		JsonObject jsonStack = NotEnoughUpdates.INSTANCE.manager.getJsonForItem(
+			NotEnoughUpdates.INSTANCE.manager.
+				createItemResolutionQuery().
+				withKnownInternalName(internal).
+				resolveToItemStack());
+		int rarity = Utils.getRarityFromLore(jsonStack.get("lore").getAsJsonArray());
+
+		if (internal.equals("HEGEMONY_ARTIFACT")) {
+			switch (rarity) {
+				case 4:
+					return 16;
+				case 5:
+					return 22;
+			}
+		}
+
+		if (internal.contains("ABICASE")) {
+			try {
+				JsonObject profileInfo = GuiProfileViewer.getSelectedProfile().getProfileJson();
+				if (profileInfo.has("nether_island_player_data")) {
+					JsonObject data = profileInfo.get("nether_island_player_data").getAsJsonObject();
+					if (data.has("abiphone") && data.get("abiphone").getAsJsonObject().has("active_contacts")) { // BatChest
+						int contact = data.get("abiphone").getAsJsonObject().get("active_contacts").getAsJsonArray().size();
+						abi = contact / 2;
+					}
+				}
+			} catch (NullPointerException e) {
+			}
+		}
+
+		switch (rarity) {
+			case 0:
+			case 6:
+				return abi + 3;
+			case 1:
+			case 7:
+				return abi + 5;
+			case 2:
+				return abi + 8;
+			case 3:
+				return abi + 12;
+			case 4:
+				return abi + 16;
+			case 5:
+				return abi + 22;
+		}
+
+		return 0;
+	}
+
+	public static void renderButton(ItemStack stack, int x, int y, List<String> tooltip) {
+		GlStateManager.color(1, 1, 1, 1);
+		Minecraft.getMinecraft().getTextureManager().bindTexture(accessory_bag_overlay);
+		GlStateManager.disableLighting();
+		Utils.drawTexturedRect(x, y, 17, 17, 168f / 196f, 184f / 196f, 112f / 128f, 1f, GL11.GL_NEAREST); // slot
+		RenderHelper.enableGUIStandardItemLighting();
+		Minecraft.getMinecraft().getRenderItem().renderItemIntoGUI(stack, x, y); // item
+		if (new Rectangle(x, y, 16, 16).contains(mouseX(), mouseY())) {
+			Utils.drawHoveringText(
+				tooltip,
+				mouseX(), mouseY(),
+				getScaledResolution().getScaledWidth(),
+				getScaledResolution().getScaledHeight(),
+				-1
+			);
+		}
+	}
+
+	public static void highlightDuplicates(int x, int y) {
+		int guiLeft = (int) Utils.getField(
+			GuiContainer.class,
+			Minecraft.getMinecraft().currentScreen,
+			"guiLeft",
+			"field_147003_i"
+		);
+		int guiTop = (int) Utils.getField(
+			GuiContainer.class,
+			Minecraft.getMinecraft().currentScreen,
+			"guiTop",
+			"field_147009_r"
+		);
```

Comment:
```suggestion
		AccessorGuiContainer accessor = (AccessorGuiContainer) Minecraft.getMinecraft().currentScreen;
		int guiLeft = accessor.getGuiLeft();
		int guiTop = accessor.getGuiTop();
```

also in the other place its used